zebra-state/src/service/arbitrary.rs

```diff
@@ -60,6 +60,30 @@ impl ValueTree for PreparedChainTree {
 pub struct PreparedChain {
     // the proptests are threaded (not async), so we want to use a threaded mutex here
     chain: std::sync::Mutex<Option<(Network, Arc<SummaryDebug<Vec<PreparedBlock>>>)>>,
+    // the height from which to start the chain. If None, starts at the genesis block
+    start_height: Option<Height>,
+}
+
+impl PreparedChain {
+    /// Create a PreparedChain strategy with Heartwood-onward blocks.
+    #[cfg(test)]
+    pub(crate) fn new_heartwood() -> Self {
+        // The history tree only works with Heartwood onward.
+        // Since the network will be chosen later, we pick the larger
+        // between the mainnet and testnet Heartwood activation heights.
+        let main_height = NetworkUpgrade::Heartwood
+            .activation_height(Network::Mainnet)
+            .expect("must have height");
+        let test_height = NetworkUpgrade::Heartwood
+            .activation_height(Network::Testnet)
+            .expect("must have height");
+        let height = (std::cmp::max(main_height, test_height) + 1).expect("must be valid");
```

Comment:
Optional: We could simplify this code by storing an "after heartwood" flag in `PreparedChain`, then adjusting the height to `Heartwood::activation_height(network)` in the `.prop_flat_map(|ledger| {` closure.

Ideally, we should make this change in the ledger state itself, but it's not a high priority right now.

Reply:
Sounds good.

Feel free to open a ticket, or we can just make the change if it ever comes up again.

Reply:
Created in https://github.com/ZcashFoundation/zebra/issues/2604